ClientApp/viking-vault/src/RegisterForm.tsx

```diff
@@ -0,0 +1,233 @@
+import React from "react";
+import {ResponseModal} from "./ReponseModal";
+import {Redirect, BrowserRouter as Router} from "react-router-dom";
+
+const url = "https://localhost:44323/api/";
+const emailRegEx = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+interface IFormState {
+    [key: string]: string | number | boolean;
+    email: string;
```

Comment:
It's good to put related object properties into their own object, it seems to me like email, password, cpassword, firstName...etc are all part of the same form and represent the same entity.

Reply:
I'm not sure I understand. It's not ok that all of them are in the same interface?

Reply:
What I mean is, that if you have related properties, let's say that you have a person and some other state properties 


```
interface IFormState {
	[key: sring]: number;
	isModalOpen: boolean;
	isLoading: boolean;
	otherComponentRelatedState: SoemOtherType;
	firstName: string;
	lastName: string;
	age: number;
}
```

you'd better have something like this 

```
interface IFormState {
	[key: sring]: number;
	isModalOpen: boolean;
	isLoading: boolean;
	otherComponentRelatedState: SoemOtherType;
	person: {
		firstName: string;
		lastName: string;
		age: number;
	}
	
}
```


or even better


```
interface IPerson {
	firstName: string;
	lastName: string;
	age: number;
}

interface IFormState {
	[key: sring]: number;
	isModalOpen: boolean;
	isLoading: boolean;
	otherComponentRelatedState: SoemOtherType;
	person: IPerson	
}
```

Reply:
Ok. I understand now. Thank you!